Compilation error redefinition of typedef 'nghttp2_data_provider_wrap' in 1.60.0 for old gcc

When compiling nghttp2 1.60.0 on old linux systems, eg RHEL 6 or SLES11, using gcc 4.4.5 or 4.3.4, I get the following fatal compilation error:

In file included from nghttp2_session.c:42:
nghttp2_submit.h:34: error: redefinition of typedef 'nghttp2_data_provider_wrap'
nghttp2_outbound_item.h:49: note: previous declaration of 'nghttp2_data_provider_wrap' was here
make[3]: *** [nghttp2_session.lo] Error 1

File nghttp2_session.c includes nghttp2_session.h, which in turn includes nghttp2_outbound_item.h which defines

 39 typedef struct nghttp2_data_provider_wrap {
 40   int version;
 41   union {
 42     struct {
 43       nghttp2_data_source source;
 44       void *read_callback;
 45     };
 46     nghttp2_data_provider v1;
 47     nghttp2_data_provider2 v2;
 48   } data_prd;
 49 } nghttp2_data_provider_wrap;

Later file nghttp2_session.c includes nghttp2_submit.h, which has

 34 typedef struct nghttp2_data_provider_wrap nghttp2_data_provider_wrap;

Both have been added recently in 93c4d8bc340438460bce20c939a357fbfae05350. Newer GCC version do not seem to complain.

Best regards,

Rainer

--- a/lib/nghttp2_submit.h
+++ b/lib/nghttp2_submit.h
@@ -31,7 +31,7 @@
 
 #include <nghttp2/nghttp2.h>
 
-typedef struct nghttp2_data_provider_wrap nghttp2_data_provider_wrap;
+#include "nghttp2_outbound_item.h"
 
 int nghttp2_submit_data_shared(nghttp2_session *session, uint8_t flags,
                                int32_t stream_id,

